### PR TITLE
Fix #17039 - abj output ##json

### LINKED ISSUE #17039: ab, abj and afbj commands
The help for the `ab` command is:
> | ab [addr]            show basic block information at given address

This command works fine, and gives information just for one basic block. But when you use `abj` to get this information in a JSON format, you'd expect to get them only for the target basic block. Instead, you get information for every basic block in the function, since as mentioned in the help, `abj` is an alias to `afbj`:
> | abj [addr]           display basic block information in JSON (alias to afbj)

This is a bit counterintuitive, because adding a  `j` to a command usually gives you the same result that the same command without it, but in a JSON format, and you now have to go through the returned list to find the basic block that you asked for.

### PR DESCRIPTION
**Pull request by mrglm**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](https://github.com/radareorg/radare2book) with the relevant information (if needed)

**Detailed description**

`abj` used to be an alias to `afbj`, but now it returns information (in JSON) for only one basic block.

I'm not sure what is the difference between  the properties `addr` and `opaddr` of a basic block, and I did not find documentation about it. Still, I think that the `opaddr`  of a block might be incorrect sometimes, so hit me up if you have information. 

**Test plan**

...

**Closing issues**

Closes #17039

### CODE REVIEW

File: libr/core/cmd_anal.c [hunk 1]
```diff
@@ -10238,7 +10241,18 @@ static int cmd_anal(void *data, const char *input) {
 		} else if (input[1] == 't') {
 			cmd_anal_abt (core, input+2);
 		} else if (input[1] == 'j') { // "abj"
-			anal_fcn_list_bb (core, input + 1, false);
+			ut64 addr = core->offset;
+            if (input[2] && input[2] != '.') {
```
Comment (ret2libc) [inline libr/core/cmd_anal.c:10245]: wrong indentation here and in all following lines.

**Comment by trufae** [comment]
Dupped work with https://github.com/radareorg/radare2/pull/17680

File: libr/core/cmd_anal.c [hunk 2]
```diff
@@ -10241,7 +10244,18 @@ static int cmd_anal(void *data, const char *input) {
...
+			if (input[2] && input[2] != '.') {
+				addr = r_num_math (core->num, input + 2);
+			}
+			RAnalBlock *b = r_anal_get_block_at (core->anal, addr);
+			RAnalFunction *fcn = r_anal_get_fcn_in (core->anal, addr, 0);
+			PJ *pj = pj_new ();
+			if (b && fcn && pj) {
+				pj = bb_to_pj (pj, b, fcn, addr);
+				r_cons_println (pj_string (pj));
+				pj_free (pj);
```
Comment (trufae) [inline libr/core/cmd_anal.c:10257]: if b or fcn are null pj will never be freed

**Review by trufae — CHANGES_REQUESTED** [review]
What's left to be done to not b e considered WIP?

**Comment by mrglm** [comment]
There was some work that was a duplicate with another pull request, so I was waiting for this PR to be merged. I'll get back to it and see if there is any work left to be done.

**Comment by trufae** [comment]
Yeah the other PR got merged, so this one must be rebased, i think moving this logic into a separate function makes sense. about your questions:

* addr = adderss of the basic block holding the current instruction
* opaddr = address of the instruction you are at.

this is.. if you are int he middle of a instruction abj may find the bb by address, and then check the Nth instruction that matches the given offset and provide that info

**Review by trufae — APPROVED** [review]

**Pull request merged**